pkg/scm/git/git.go

```diff
@@ -379,6 +395,33 @@ func doesExist(dir string) bool {
 	return !(err != nil && os.IsNotExist(err))
 }
 
+// isEmptyGitRepository checks if there are any commits or remotes tracked in
+// the repository, returns true if it is empty
+func isEmptyGitRepository(dir string) bool {
+	var isEmpty = true // The return value, true for empty
+
+	gitDir := strings.TrimPrefix(dir, "file://") + "/.git"
+	// Search the content of the .git directory for content
+	directories := [2]string{gitDir + "/objects", gitDir + "/refs"}
```

Comment:
BTW, instead of prepending `gitDir` to each element of the array here, we can prepend it later, when we're passing first argument to `filepath.Walk()`. It's up to you, of course, it's just idea.